src/main/java/reposense/parser/CsvParser.java

```diff
@@ -36,20 +36,25 @@
             + "Content: %s\n"
             + "Error: %s";
     private static final String MESSAGE_EMPTY_CSV_FORMAT = "The CSV file, %s, is empty.";
+    private static final String MESSAGE_WRONG_HEADER_SIZE = "Wrong number of columns in header of CSV file, %s. \n"
+        + "Number of columns in header: %d\n"
+        + "Expected number of columns: %d";
 
     private Path csvFilePath;
+    private int expectedHeaderSize;
     private int numOfLinesBeforeFirstRecord = 0;
 
     /**
      * @throws IOException if {@code csvFilePath} is an invalid path.
      */
-    public CsvParser(Path csvFilePath) throws IOException {
+    public CsvParser(Path csvFilePath, int expectedHeaderSize) throws IOException {
```

Comment:
@fzdy1914, @jylee-git, do you think we can could keep the expected header size as a private variable in `CsvParser` like this?

With this approach, each child just needs to pass in the correct value of `expectedHeaderSize` when calling the constructor of the superclass.

Reply:
I think it is a good approach. Do request a review from us again after you have finished everything.